### PR TITLE
{chem}[foss/2020a,fosscuda/2020a] Amber v20.11-AmberTools-20.15 w/ Python 3.8.2

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

Requires:
* [x] EasyBlock update: https://github.com/easybuilders/easybuild-easyblocks/pull/2445
* [x] PnetCDF (gompic): https://github.com/easybuilders/easybuild-easyconfigs/pull/12986

### CODE REVIEW

File: easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name = 'Amber'
+local_amber_ver = 20
+local_ambertools_ver = 20
+# Patch levels from http://ambermd.org/AmberPatches.php and http://ambermd.org/ATPatches.php
+patchlevels = (15, 11)  # (AmberTools, Amber)
+version = '%s.%s-AmberTools-%s.%s' % (local_amber_ver, patchlevels[1], local_ambertools_ver, patchlevels[0])
+versionsuffix = '-Python-%(pyver)s'
+
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb:8]: I did:
```
local_amber_ver = '18'
local_ambertools_ver = '19'
# Patch levels from http://ambermd.org/bugfixes16.html and http://ambermd.org/bugfixesat.html
patchlevels = (12, 17)  # (AmberTools, Amber)
local_AT_suffix = '-AmberTools-%s.%s' % (local_ambertools_ver, patchlevels[0])

version = '18.%s' % patchlevels[1]
versionsuffix = '%s-Python-%%(pyver)s' % local_AT_suffix
```

Comment (akesandgren) [reply easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb:8]: I think that is a more correct version setting...

File: easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+homepage = 'http://ambermd.org/amber.html'
+description = """Amber (originally Assisted Model Building with Energy Refinement) is software for performing
+ molecular dynamics and structure prediction."""
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+# Amber CMake fails when 'usempi': True is set and we add '-DMPI=TRUE'
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb:14]: Then we might need to wrap up a patch to fix that, and send it upstream...

Comment (branfosj) [reply easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb]: The error I am seeing is
```
-- Found MPI_C: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenMPI/4.0.3-GCC-9.3.0/bin/mpicc (found version "3.1") 
-- Found MPI_CXX: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenMPI/4.0.3-GCC-9.3.0/bin/mpicxx (found version "3.1") 
-- Found MPI_Fortran: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenMPI/4.0.3-GCC-9.3.0/bin/mpifort (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- MPI C Compiler: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenMPI/4.0.3-GCC-9.3.0/bin/mpicc
-- MPI CXX Compiler: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenMPI/4.0.3-GCC-9.3.0/bin/mpicxx
-- MPI Fortran Compiler: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenMPI/4.0.3-GCC-9.3.0/bin/mpifort
-- If these are not the correct MPI wrappers, then set MPI_<language>_COMPILER to the correct wrapper and reconfigure.
CMake Error at cmake/LibraryTracking.cmake:174 (message):
  Incorrect usage.  At least one LIBRARY should be provided.
Call Stack (most recent call first):
  cmake/MPIConfig.cmake:122 (import_libraries)
  CMakeLists.txt:118 (include)
```

Comment (branfosj) [reply easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb]: This https://gitlab.kitware.com/cmake/cmake/-/issues/19427#note_589242 looks like issue - that `MPI_C_LIBRARIES` (and CXX and Fortan) are not set, but Amber expects them to be.

File: easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+dependencies = [
+    ('Boost.Python', '1.72.0'),
+    ('bzip2', '1.0.8'),
+    ('libreadline', '8.0'),
+    ('matplotlib', '3.2.1', '-Python-%(pyver)s'),
+    ('netCDF-Fortran', '4.5.2'),
+    ('netCDF', '4.7.4'),
+    ('Perl', '5.30.2'),
+    ('PnetCDF', '1.12.1'),
+    ('Python', '3.8.2'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb:40]: I think it's better to have Python before Boost.Python

File: easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+toolchainopts = {'usempi': True, 'openmp': True}
+
+sources = [
+    '%%(name)s%s.tar.bz2' % local_amber_ver,
+    'AmberTools%s.tar.bz2' % local_ambertools_ver,
+]
+patches = [
+    '%%(name)s-%s_cmake-locate-netcdf.patch' % local_amber_ver,
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb:21]: This is an AmberTools patch, not Amber, and should thus be named AmberTools-20_cmake... to make it easier to have a separate AmberTools package for those who don't want to buy Amber

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb (hunk 2 above)
Comment (branfosj) [reply easybuild/easyconfigs/a/Amber/Amber-20.11-AmberTools-20.15-foss-2020a-Python-3.8.2.eb]: Fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/2

File: easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb [hunk 5]
```diff
@@ -18,7 +18,7 @@ sources = [
     'AmberTools%s.tar.bz2' % local_ambertools_ver,
 ]
 patches = [
-    '%%(name)s-%s_cmake-locate-netcdf.patch' % local_amber_ver,
+    'AmberTools-%s_cmake-locate-netcdf.patch' % local_amber_ver,
```
Comment (akesandgren) [inline easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb:21]: Now it should be local_ambertools_ver

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb (hunk 5 above)
Comment (branfosj) [reply easybuild/easyconfigs/a/Amber/Amber-20.11-foss-2020a-AmberTools-20.15-Python-3.8.2.eb]: changed in ee9762f

**Comment by akesandgren** [comment]
Test report by @akesandgren
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2445
**FAILED**
Build succeeded for 4 out of 6 (2 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.5
See https://gist.github.com/9418cbf132737d9023781339dd8d94b6 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2445
**FAILED**
Build succeeded for 1 out of 3 (2 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.5
See https://gist.github.com/b1055d4f0525eb8471c70bdbe80fd098 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2445
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.5
See https://gist.github.com/b43ebf3376d48f8ee567b156f14f5145 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2445
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u14b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/fea8510fa38df2d2056c95691b3e4d19 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0306u03a.bear.cluster - Linux RHEL 8.4, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/cd6b3022d25c841b8bd908c1dad1c486 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @branfosj!

**Pull request merged**